app/controllers/dashboard_controller.rb

```diff
@@ -2,54 +2,45 @@ class DashboardController < ApplicationController
   before_action :authenticate_user!
   before_action :logged_in_user
   include DashboardHelper
+
   def index
     today = date_to_url(DateTime.now)
     redirect_to dashboard_path(today)
   end
   def show
     @chosen_date = params[:id].to_date
-    @query_dates = Income.select(:date).distinct.limit(5).order('date DESC').reverse
-    @dates = Array.new()
-    @query_dates.each do |item|
-      @dates << { active: is_chosen_date?(item.date, @chosen_date), 
-                  date_name: name_of_day(item.date),
-                  date_value: item.date.to_s(:db) 
-                }
-    end
-    @general_expenses = GeneralExpense.list.where(date: @chosen_date).decorate
-    @incomes_truck_routes = Income.truck_routes.where(date: @chosen_date).decorate
-    @incomes_brand_shops = Income.brand_shops.where(date: @chosen_date).decorate
-    @incomes_external_shops = Income.external_shops.where(date: @chosen_date).decorate
+    last_workdays_data(@chosen_date)
+    fetch_data(@chosen_date)
     @chosen_date = @chosen_date.strftime("%Y-%m-%d, %A")
```

Comment:
date formats should end up in locales and you can easly use them with `to_formatted_s(:format_name)` method